tap.d.ts

```diff
@@ -0,0 +1,39 @@
+// tap.d.ts
+
+declare module Test {
+
+  class Deferred {
+    resolve();
+    reject();
+  }
+
+  class test {
+    constructor(options?: any);
+    tearDown();
+    setTimeout(n: number);
+    endAll();
+    threw(er: Error, extra: Error, proxy: any);
+    pragma(set: any);
+    plan(n: number, comment?: string);
+    test(name: string, extra: Error, cb: () => void, deferred: Deferred);
+    loop(self: test, arr: test[], cb: () => void, i: number);
+    next(er: Error);
+    current();
+    stdin(name: string, extra: any, deferred: Deferred);
+    spawn(cmd: string, args: string, options: any, name: string, extra: any, deferred: Deferred);
+    done(implicit: boolean);
+    autoend();
+    printResult(ok: boolean, message: string, extra: any);
+    writeDiags(extra: any);
```

Comment:
These should only be used internally, but they're (perhaps incorrectly) exposed on the class.  Maybe we ought to just have them accept a `self` argument, and be hidden?